src/middleware/TelemetryHandler.ts

```diff
@@ -64,23 +65,44 @@ export class TelemetryHandler implements Middleware {
 	 * @param {Context} context - The context object of the request
 	 * @returns A Promise that resolves to nothing
 	 */
+
+	/**
+	 * @private
+	 * Checks if the request url points to the Graph service endpoints
+	 * @param {string} url - The request url string
+	 * @returns true if request url is a Graph url
+	 */
+	private isGraphURL(url: string): boolean {
```

Comment:
1. construct a `Set()` object for `graphURLs` where the values are only the host names.
2. construct a typed `URL` object for `url`.
3. Have this function return the result of `graphURLs.has(urlObject.hostname)`

Benefits:
1. Efficiency: Lookup is amortized O(1).
2. Correctness: We probably want to capture telemetry for urls like: `https://graph.microsoft.com:443` and current implementation will not accept URLs with port numbers.

Reply:
also consider inlining the whole function as it will be a simple call to `graphURLs.has(new URL(context.request).hostname)`

Reply:
Thank you for suggesting an optimal approach!

I have used the approach you suggested with some modifications. Implemented a set object with the host names. I haven't used the URL object as  some browsers do not support of the URL object and I am extracting the host name as it is done in [GraphRequest.ts](https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/66a41814cace93cf1da8d5ead69b8df86c1881b0/src/GraphRequest.ts#L135) during the url parsing process.